docs/lifecycle-management.md

```diff
@@ -0,0 +1,185 @@
+# Higher level VM Management
+
+## Motivation
+
+So far KubeVirt provides a pod-like interface when managing VMs. This seems to
+be a good building block.
+On the other hand this might not be sufficient for usual virtualization
+management UIs, which do expect to manage the complete life-cycle of a VM,
+including shut down VMs 8which is currently not intended by the core runtime).
+
+
+This document outlines how minimal assumptions can be met and also visits how
+other requirements impact the design.
+
+
+## Requirements
+* Running and non-running VMs are handled through the same API
+* Creating a VM must be separate from launching a VM
+* Support for “flavor” based VMs (templating)
+* Support for expanding sparse VM definitions
+* Support for detailed VM definitions
+* Support hot-plugging
+* Representation of dynamic instance specific data
+* Support for defaulting (to support OS compatibility)
+Considerations
+* Sub-resources for actions
+* Presets to add certain configurations
+
+
+## Definitions
+* **Defaulting** - Setting certain defaults during expansion (i.e. virtio for a
+  bus type, or e1000 for as a nic model)
+* **Expansion** - VM definitions can be provided in a sparse way i.e. just a
+  disk. Once the VM is getting started, the runtime will infer additional
+  requirements for this device and add them to the runtime VM definition.
+  This process of inferring and adding additional devices is called expansion.
+* **Flavor** - Comparable to a template or a specific instance type
+* **Preset** - Presetting certain values in a specified path of the document
+  (i.e. setting .spec.domain.devices.nvram). It is not about setting certain
+  default values for undefined fields (i.e. set virtio for all busses)
```

Comment:
There are two parts to libosinfo database. First there is the recording of what is supported on a technical basis. So this for example, says RHEL7 can support vga, cirrus, qxl, etc.  Second there is a concept of a 'deployment' which encodes a preferred set of devices, which is a subset of the first set. Unfortunately, while the latter concept exists at the API level, we've never provided any database records behind it, so its mostly only useful in theory, not practice. The way virt-manager deals with this though, is that it simply has its own preferred set of devices. eg virt-manager prefers qxl, cirrus, vga. So if the libosinfo reports support for multiple graphics drivers they'll be picked in that order. In practice this has worked sufficiently well that the libosinfo deployment concept has not been needed. I think it'd work fine for kubevirt too.

@fabiand that is pretty much exactly how I would expect it to work. I would probably expect the os distro to be listed as an annotation on the VM spec TPR, not part of the main config description